Sources/Swift/Integrations/SessionReplay/SentryBreadcrumbReplayConverter.swift

```diff
@@ -0,0 +1,83 @@
+@_implementationOnly import _SentryPrivate
+import Foundation
+
+@objcMembers
+class SentryBreadcrumbReplayConverter: NSObject {
+    
+    private let supportedNetworkData = Set<String>([
+        "status_code",
+        "method",
+        "response_content_length",
+        "request_content_length",
+        "http.query",
+        "http.fragment"]
+    )
+    
+    func replayBreadcrumbs(from breadcrumbs: [Breadcrumb]) -> [SentryRRWebEvent] {
+        breadcrumbs.compactMap { replayBreadcrumb(from: $0) }
+    }
+    
+    //Convert breadcrumb information into something
+    //replay front understands
+    private func replayBreadcrumb(from breadcrumb: Breadcrumb) -> SentryRRWebEvent? {
+        guard let timestamp = breadcrumb.timestamp else { return nil }
+        if breadcrumb.category == "http" {
+            return networkSpan(breadcrumb)
+        } else if breadcrumb.type == "navigation" {
+            return navigationBreadcrumb(breadcrumb)
+        } else if breadcrumb.category == "touch" {
+            return SentryRRWebBreadcrumbEvent(timestamp: timestamp, category: "ui.tap", message: breadcrumb.message)
+        } else if breadcrumb.type == "connectivity" && breadcrumb.category == "device.connectivity" {
+            guard let networkType = breadcrumb.data?["connectivity"] as? String, !networkType.isEmpty  else { return nil }
+            return SentryRRWebBreadcrumbEvent(timestamp: timestamp, category: "device.connectivity", data: ["state": networkType])
+        } else if let action = breadcrumb.data?["action"] as? String, action == "BATTERY_STATE_CHANGE" {
+            var data = breadcrumb.data?.filter({ item in item.key == "level" || item.key == "plugged" }) ?? [:]
+            
+            data["charging"] = data["plugged"]
+            data["plugged"] = nil
```

Comment:
`l`: Please comment on why we set this to `nil`. I don't get it.

Reply:
Removing information that will not be used.

Reply:
I'm sorry I haven't been clear enough. I meant, please this comment to the code.